src/components/UserList.jsx

```diff
@@ -1,4 +1,42 @@
+import { useContext } from "react";
+import userContext from "../store/userContext";
+import SyncLoader from "react-spinners/SyncLoader";
+
 const UserList = () => {
-  return <div>UserList</div>;
+  const { users, loginUser, setLoginUser, isLoading } = useContext(userContext);
+
+  return (
+    <>
+      {isLoading ? (
+        <div className="overlay">
+          <SyncLoader className="spinner" />
+        </div>
+      ) : (
+        <>
+          <section className="userList__container">
+            {users.map((user) => (
+              <div
+                key={user.username}
+                className="user__container"
+                onClick={() => setLoginUser(user.username)}
+              >
+                <h2>{user.name}</h2>
+                <p>Username: {user.username}</p>
+                <img src={user.avatar_url} alt={user.username} />
+              </div>
```

Comment:
I think extracting this into a UserCard component to de-bloat the component would be ideal